tests/functional/graphql/test_get_slate_lineup.py

```diff
@@ -0,0 +1,84 @@
+from graphql.execution.executors.asyncio import AsyncioExecutor
+from graphene.test import Client
+from fastapi.testclient import TestClient
+
+from app.graphql.graphql import schema
+from app.main import app, load_slate_configs
+from tests.functional.test_dynamodb_base import TestDynamoDBBase
+
+from unittest.mock import patch
+from collections import namedtuple
+
+MockResponse = namedtuple('MockResponse', 'status')
+
+class TestGetSlateLineup(TestDynamoDBBase):
+    client: Client
+
+    async def asyncSetUp(self):
+        await super().asyncSetUp()
+        self.populate_candidate_sets_table()
+        self.client = Client(schema)
+
+    @patch('aiohttp.ClientSession.get', to_return=MockResponse(status=200))
+    def test_get_slate_lineup(self, mock_clientsession_get):
+        with TestClient(
+                app):  # This context manager forces the FastAPI startup event to run, which we use to populate our slate lineup configuration
+            executed = self.client.execute(
+                '''
+                    query GetSlateLineup {
+                      getSlateLineup(slateLineupId: "aaf47c0f-1361-4c8c-a89f-fa45e1dc2978") {
+                        slates {
+                            description
+                        }
+                      }
+                    }
+                ''',
+                context_value={"user_id": "johnjacobjingleheimerschmidt"},
+                executor=AsyncioExecutor())
+
+            # No get defaults here. This chain of 'gets' is deliberately not robust.
+            # We want to test that THIS structure exists in the JSON in its entirety
+            # because our clients depend on this structure.
+            slate_list = executed.get('data').get('getSlateLineup').get('slates')
+
+            # Testing that multiple slates come back
+            assert len(slate_list) > 1
+
+            # Testing that they all have a description
+            assert all('description' in slate for slate in slate_list)
+
+            # FOR YOUR REFERENCE, the response JSON should look something like this:
+            # {'data':
+            #     {'getSlateLineup':
+            #         {'slates': [
+            #             {'description': 'Stories to fuel your mind, curated for your interests'},
+            #             {'description': 'Pocket Collections'},
+            #             {'description': 'Smart, quick reads from trusted sources'},
+            #             {'description': 'The best longform journalism and essays popular in Pocket'},
+            #             {'description': 'Great stories that stand the test of time'}
+            #         ]
+            #         }
+            #     }
+            # }
+
+    def populate_candidate_sets_table(self):
+        self.candidate_set_table.put_item(Item={
+            'id': '303174fc-a9ff-4a51-984a-e09ce7120d18',
```

Comment:
Its interesting seeing these seeds. It looks like in a lot of ways we are testing the config. Because these lists are define elsewhere. Testing is always weird where some data is in config and the other is in the database. Out of scope of this PR, it makes me wonder if we should be looking more towards a soft config and moving slates and slate lineups configs to db and treating them as more mutable values depending on their environment.

Reply:
I think that might make sense. I'd love for us to do what we can to treat "stuff that's likely to change" as data and "stuff that's likely to not change" as code/infra, to the degree we can.

Reply:
Given that this is a very high-level API test, of which there won't be many, I think it will be easy to change the test if the lineup in question changes. So I like the approach taken here.

Tests in [TestSlateLineupModel](https://github.com/Pocket/recommendation-api/blob/main/tests/functional/models/test_slate_lineup_model.py#L94-L95) patch the config to create fixtures for all of the data stored in JSON configs, such that tests will pass even when slates and lineups change. The way fixtures are defined in TestSlateLineupModel (60 lines of code at the top of the file) makes the code hard to understand and change, but this could be improved by using a library for organizing fixtures such as [PyTest fixtures](https://docs.pytest.org/en/latest/explanation/fixtures.html).